portal-kernel/src/com/liferay/portal/kernel/servlet/ServletOutputStreamAdapter.java

```diff
@@ -38,6 +39,21 @@ public void flush() throws IOException {
 		outputStream.flush();
 	}
 
+	/**
+	* @since Servlet 3.1
+	*/
+	@Override
+	public boolean isReady() {
+		return true;
+	}
+
+	/**
+	* @since Servlet 3.1
+	*/
+	@Override
+	public void setWriteListener(WriteListener listener) {
+	}
+
```

Comment:
@holatuwol The requirements from the [Javadoc for this method](https://docs.oracle.com/javaee/7/api/javax/servlet/ServletOutputStream.html#setWriteListener-javax.servlet.WriteListener-) have not been met either.